src/mechanics/cell/Cell.h

```diff
@@ -1,89 +1,85 @@
 #pragma once
 
-#include "mechanics/cell/CellInterface.h"
 #include <boost/ptr_container/ptr_vector.hpp>
-#include "mechanics/elements/ElementInterface.h"
+#include "mechanics/cell/CellInterface.h"
+#include "mechanics/elements/Element.h"
 #include "mechanics/nodes/DofContainer.h"
-#include "mechanics/cell/Integrand.h"
+#include "mechanics/integrands/Base.h"
 #include "mechanics/integrationtypes/IntegrationTypeBase.h"
+#include "mechanics/cell/CellData.h"
+#include "mechanics/cell/CellIpData.h"
 
 namespace NuTo
 {
 class Cell : public CellInterface
 {
 public:
-    Cell(const ElementInterface& coordinateElement, DofContainer<ElementInterface*> elements,
-         const IntegrationTypeBase& integrationType, const Integrand& integrand)
-        : mCoordinateElement(coordinateElement)
-        , mElements(elements)
+    Cell(const Element& element, const IntegrationTypeBase& integrationType, const Integrand::Base& integrand)
+        : mElement(element)
         , mIntegrationType(integrationType)
-        , mIntegrand()
+        , mIntegrands()
     {
         for (int i = 0; i < integrationType.GetNumIntegrationPoints(); i++)
-            mIntegrand.push_back(integrand.Clone());
+            mIntegrands.push_back(integrand.Clone().release());
     }
 
-    //! @brief builds the internal gradien
-    DofVector<double> Gradient() override
+    DofVector<double> operator()(const VectorOperation& op) override
```

Comment:
I think these should be called `Integrate`, because that's what they do. Calling `myCell(someOperation)` does not simply "do the operation", but integrates it.

Reply:
Ok, then we would have `Integrate` for `Scalar/Vector/MatrixOperation` and `Apply` for `VoidOperation`? Or better keep `operator()` for the `VoidOperation`?